src/Nethermind/Nethermind.Blockchain/Receipts/ReceiptsRootCalculator.cs

```diff
@@ -0,0 +1,43 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Specs;
+using Nethermind.Serialization.Rlp;
+using Nethermind.State.Proofs;
+
+namespace Nethermind.Blockchain.Receipts;
+
+public class ReceiptsRootCalculator : IReceiptsRootCalculator
+{
+    public static readonly ReceiptsRootCalculator Instance = new();
+
+    public Hash256 GetReceiptsRoot(TxReceipt[] receipts, IReceiptSpec spec, Hash256? suggestedRoot)
+    {
+        Hash256 SkipStateAndStatusReceiptsRoot()
+        {
+            receipts.SetSkipStateAndStatusInRlp(true);
```

Comment:
not directly related, but could we refactor this away by any chance?

Reply:
Honestly, I have no idea what is that